src/Sarif/Autogenerated/IRule.cs

```diff
@@ -10,8 +11,8 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// <summary>
     /// Interface exposed by objects that provide information about analysis rules.
     /// </summary>
-    [GeneratedCode("Microsoft.Json.Schema.ToDotNet", "0.22.0.0")]
-    public interface IRule
+    [GeneratedCode("Microsoft.Json.Schema.ToDotNet", "0.28.0.0")]
+    public partial interface IRule
```

Comment:
The SDK includes code in which an `IRule`-derived object needs access to the properties (below, you'll see a line of code that references `iRule.Properties`). But `Properties` can't be part of the public interface. What we want instead is for the property accessor methods `{Get,Set}Property` to be part of the `IRule` interface. But the code generator can't generate those methods, which are specific to the SDK.

So I modified the code generator to emit _partial_ interfaces. Now the client of the generator -- the SDK -- can add whatever it wants to the generated interfaces. You'll see below that `IRule` now extends `IPropertyBagHolder`, which in turn declares the property accessor methods.